verilog/analysis/checkers/module_filename_rule.cc

```diff
@@ -73,6 +74,20 @@ void ModuleFilenameRule::Lint(const TextStructureView& text_structure,
   // Find all module declarations.
   auto module_matches = FindAllModuleDeclarations(*tree);
 
+  // If there are no modules in this source unit, suppress finding.
+  if (module_matches.empty()) return;
+
+  // Remove nested module declarations
+  std::vector<verible::TreeSearchMatch> module_cleaned;
+  module_cleaned.reserve(module_matches.size());
+  std::back_insert_iterator<std::vector<verible::TreeSearchMatch>> back_it(module_cleaned);
+  std::remove_copy_if(module_matches.begin(),
+                      module_matches.end(),
+                      back_it,
```

Comment:
FYI, can replace `back_it` with `std::back_inserter(module_cleaned)` (easier to write/read).